addon/model/model-range.ts

```diff
@@ -259,6 +262,63 @@ export default class ModelRange {
     return this.start.sameAs(this.end);
   }
 
+  normalize(): void {
+    let flip = false;
+    const length =
+      this.start.path.length > this.end.path.length
+        ? this.start.path.length
+        : this.end.path.length;
+
+    for (let i = 0; i < length; i++) {
+      if (this.start.path[i] == undefined || this.end.path[i] == undefined) {
+        break;
+      } else if (this.start.path[i] == this.end.path[i]) {
+      } else if (this.start.path[i] > this.end.path[i]) {
+        flip = true;
+        break;
+      } else if (this.start.path[i] < this.end.path[i]) {
+        break;
+      }
+    }
+
+    if (flip) {
+      const buffer = this.start;
+      this.start = this.end;
+      this.end = buffer;
+    }
+  }
+  /**
+   * janky debug function
+   */
+  visualize(): string {
+    let root = this.root;
+    while (root.parent) {
+      root = root.parent;
+    }
+    root = root.clone();
+    const range = ModelRange.fromPaths(root, this.start.path, this.end.path);
+
+    const startRange = new ModelRange(range.start, range.start);
+    const endRange = new ModelRange(range.end, range.end);
+    const startText = new ModelText('[===START===]');
+    const endText = new ModelText('[===END===]');
+
+    OperationAlgorithms.insert(endRange, endText);
```

Comment:
This breaks a big assumption of operations being the smallest unit of state changes, please use at least an operation. You can pass undefined for the event-bus parameter. (you're working on a copy, so technically it's fine to use the algorithm directly, but it sets a bad precedent.